### PR TITLE
PC-1119 Success Page Changes to Wording and Logic

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1119)

# Description

changes the success page & referral email to the new requested format

adds a new context variable that controls whether the section on ECO4 is shown

the implied previous format from the provided client document did not match the current webpage. to keep the situation aligned to client expectations, the text under the 'What happens next?' header follows the document format exactly as there were many differences present

text under other headers received the specified edits only, as there were only minor differences in client expectation

translations are still pending

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVK4AgQ3E=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

# Screenshots

image
image

### CODE REVIEW

File: help_to_heat/frontdoor/views.py [hunk 1]
```diff
@@ -921,8 +921,17 @@ def handle_post(self, request, session_id, page_name, data, is_change_page):
...
         supplier = SupplierConverter(session_id).get_supplier_on_success_page()
+        session_data = interface.api.session.get_session(session_id)
+        benefits = session_data.get("benefits")
+        household_income = session_data.get("household_income")
+        is_eco4_eligible = benefits == "Yes" or household_income == "Less than £31,000 a year"
         referral = portal.models.Referral.objects.get(session_id=session_id)
-        return {"supplier": supplier, "safe_to_cache": True, "referral_id": referral.formatted_referral_id}
+        return {
+            "supplier": supplier,
+            "safe_to_cache": True,
```
Comment (jdgage) [inline help_to_heat/frontdoor/views.py:931]: I think we may not want this now? Given the response now contains data drawn from the user's salary/benefits answer and judging by the comment  where we use `safe_to_cache`. Maybe something to ask for wider opinions about: I'm not sure if we necessarily need to be as strict as we're currently being.

**Review by jdgage — CHANGES_REQUESTED** [review]
Think we want a test or two covering the ECO4 text at least

File: help_to_heat/frontdoor/views.py (hunk 1 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:931]: considering that this page has always been dynamic (it contains the referral code) it seems it hasn't been safe to cache in some time. this is also the only time in the project this is used so can probably be removed. can ask Sam to confirm?

Comment (jdgage) [reply help_to_heat/frontdoor/views.py:931]: Judging by the comment it's less about the page being dynamic and more about it containing potentially sensitive data. [Looks like](https://beisdigital.atlassian.net/browse/PC-359) it's something that came out of the pen test around go-live last year. Suspect we should stick to that and remove this line. Would keep the other bit of code though.

Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:931]: have removed the safe to cache line

**Review by jdgage — APPROVED** [review]
Pending translations

**Review by jdgage — APPROVED** [review]

**Pull request merged**